streams-bootstrap-core/src/main/java/com/bakdata/kafka/consumer/ConfiguredConsumerApp.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2025 bakdata
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.bakdata.kafka.consumer;
+
+import com.bakdata.kafka.ConfiguredApp;
+import com.bakdata.kafka.EnvironmentKafkaConfigParser;
+import com.bakdata.kafka.KafkaPropertiesFactory;
+import com.bakdata.kafka.RuntimeConfiguration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import lombok.NonNull;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+
+/**
+ * A {@link ConsumerApp} with a corresponding {@link ConsumerAppConfiguration}
+ *
+ * @param <T> type of {@link ConsumerApp}
+ */
+public record ConfiguredConsumerApp<T extends ConsumerApp>(@NonNull T app,
+                                                           @NonNull ConsumerAppConfiguration configuration)
+        implements ConfiguredApp<ExecutableConsumerApp<T>> {
+    private static Map<String, Object> createBaseConfig() {
+        final Map<String, Object> kafkaConfig = new HashMap<>();
+
+        kafkaConfig.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        kafkaConfig.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
```

Comment:
Is the goal here to enable exactly-once semantics? If yes, we should add `isolation.level: read_committed` so that we don't read any uncommitted records from upstream Streams apps: https://docs.confluent.io/platform/current/installation/configuration/consumer-configs.html#isolation-level